sql/2020/01_CSS/vendor_prefix_summary.sql

```diff
@@ -0,0 +1,113 @@
+#standardSQL
+CREATE TEMPORARY FUNCTION getPrefixStats(css STRING) RETURNS ARRAY<STRING> LANGUAGE js AS '''
+try {
+	function compute() {
+		let ret = {
+			total: {},
+			pseudo_classes: {},
+			pseudo_elements: {},
+			properties: {},
+			functions: {},
+			keywords: {},
+			media: {}
+		};
+
+		ret.total = Object.fromEntries(Object.keys(ret).map(k => [k, 0]));
+
+		walkRules(ast, rule => {
+			// Prefixed pseudos
+			if (rule.selectors) {
+				let pseudos = rule.selectors.flatMap(r => r.match(/::?-[a-z]+-[\\w-]+/g) || []);
+
+				for (let pseudo of pseudos) {
+					let type = "pseudo_" + (pseudo.indexOf("::") === 0? "elements" : "classes");
+					incrementByKey(ret[type], pseudo);
+					ret.total[type]++;
+				}
+			}
+
+			if (rule.declarations) {
+				walkDeclarations(rule, ({property, value}) => {
+					// Prefixed properties
+					if (/^-[a-z]+-.+/.test(property)) {
+						incrementByKey(ret.properties, property);
+						ret.total.properties++;
+					}
+
+					// -prefix-function()
+					for (let call of extractFunctionCalls(value, {names: /^-[a-z]+-.+/})) {
+						incrementByKey(ret.functions, call.name);
+						ret.total.functions++;
+					}
+
+					// Prefixed keywords
+					if (!matches(property, /(^|-)(transition(-property)?|animation(-name)?)$/)) {
+						for (let k of value.matchAll(/(?<![-a-z])-[a-z]+-[a-z-]+(?=$|\\s|,|\\/)/g)) {
+							incrementByKey(ret.keywords, k);
+							ret.total.keywords++;
+						}
+					}
+				});
+			}
+
+			// Prefixed media features
+			if (rule.media) {
+				let features = rule.media
+									.replace(/\\s+/g, "")
+									.match(/\\(-[a-z]+-[\\w-]+(?=[:\\)])/g);
+
+				if (features) {
+					features = features.map(s => s.slice(1));
+
+					for (let feature of features) {
+						incrementByKey(ret.media, feature);
+						ret.total.media++;
+					}
+				}
+			}
+		});
+
+		ret.total.total = sumObject(ret.total);
+
+		for (let type in ret) {
+			ret[type] = sortObject(ret[type]);
+		}
+
+		return ret;
+	}
+
+
+	let ast = JSON.parse(css);
+	let stats = compute(ast).total;
+	return Object.entries(stats).flatMap(([prop, freq]) => {
+		return Array(freq).fill(prop);
+	});
+}
+catch (e) {
+	return [];
+}
+'''
+OPTIONS (library="gs://httparchive/lib/css-utils.js");
+
+SELECT
+  *
+FROM (
+  SELECT
+    client,
+    prop,
+    COUNT(DISTINCT page) AS pages,
+    COUNT(0) AS freq,
+    SUM(COUNT(0)) OVER (PARTITION BY client) AS total,
```

Comment:
This double-counts the `total` "property". See https://github.com/HTTPArchive/almanac.httparchive.org/pull/1577/files/4c28d6db9fb721f397f1600ea9a359f9b83e43c3#diff-c1fb1712782c56177cb3b4af70616e7bdce3ea485c02d97fb889301bc4d2413eR47 for an example of how to omit it from the sum of frequencies.

Reply:
Why not just divide by 2?

Reply:
It may not be clear to someone reading these queries, but omitting the magic total property from the total calculation would help to explain the intention.

Reply:
PTAL this is the last blocker